test/Fresh.hs

```diff
@@ -10,35 +10,36 @@ import Control.Effect.Fresh
 import Gen
 import qualified Hedgehog.Range as R
 import qualified Monad
+import qualified MonadFix
 import Test.Tasty
 import Test.Tasty.Hedgehog
 
 tests :: TestTree
 tests = testGroup "Fresh"
-  [ test "FreshC" FreshC.runFresh
+  [ testGroup "FreshC" $
+    [ testMonad
+    , testMonadFix
+    , testFresh
+    ] >>= ($ RunS FreshC.runFresh)
   ] where
-  test :: Has Fresh sig m => String -> (forall a . Int -> m a -> PureC (Int, a)) -> TestTree
-  test name run = testGroup name
-    $  Monad.test (m gen) a b c ((,) <$> n <*> pure ()) (uncurry run)
-    ++ Fresh.test (m gen) a                             run
+  testMonad    run = Monad.test    (m gen) a b c (atom "(,)" (,) <*> n <*> unit) run
+  testMonadFix run = MonadFix.test (m gen) a b   (atom "(,)" (,) <*> n <*> unit) run
```

Comment:
We get nice showable (albeit prefix—there’s no distfix constructor for `Term` and I didn’t feel like writing _n_ of them!) input states for the `Monad` & `MonadFix` tests now.